spec/models/piece_spec.rb

```diff
@@ -1,5 +1,226 @@
 require 'rails_helper'
 
 RSpec.describe Piece, type: :model do
-  pending "add some examples to (or delete) #{__FILE__}"
+  describe "is_obstructed? returns false if path is not blocked" do
```

Comment:
very thorough testing. you may or may not need to every permutation. that's a judgement call.

I would encourage you to add a test where the move is invalid (not horiz, vert or diag). (1,1) to (4,3), for example. what does is_obstructed do then?

Other ideas

* move to same spot (1,1) to (1,1)
* move off the board (1,1) to (100, 3)